packages/ux-capture/src/Zone.js

```diff
@@ -19,7 +39,36 @@ function Zone(props) {
 	this.measureName = this.props.name;
 
 	// Create a new `ExpectedMark` for each mark
-	this.marks = this.props.marks.map(markName => {
+	const configuredMarkNames = this.props.elements
+		? // new elements array on zone object
+		this.props.elements.map(element => element.marks).flat()
+		: // legacy with direct marks array on zone object
+		this.props.marks;
+
+	// mark names for elements already on the page after interactive transition
+	let elementsAlreadyOnThePage = [];
+
+	if (this.props.startMarkName === INTERACTIVE_TRANSITION_START_MARK_NAME &&
+		this.props.elements &&
+		Array.isArray(this.props.elements)
+	) {
+		elementsAlreadyOnThePage = this.props.elements.filter(element => {
+			const nodes = this.selectDOMNodes(element);
+
+			// if array or element list is returned, check if it has one or more entries
+			return (nodes && (typeof nodes.length === 'undefined' || nodes.length > 0));
+		});
+	}
+
+	// do not create marks for elements that are already on the page
```

Comment:
why don't we have to do this filtering for the legacy-style marks as well?

Reply:
Because it is expected that instrumentation code would call `clearMark()` for marks that represent elements that are removed from the page.

So if all the marks that are still on timeline but were recorded before zone's `startTransition` mark are automatically considered resolved later when zone is measured. See the runtime logic for it here: https://github.com/ux-capture/ux-capture/blob/a58ab0bcbf1db8fdde368e96efa7ccbdd2c7b89b/packages/ux-capture/src/Zone.js#L144-L164